src/source/Common/AwsV4Signer.c

```diff
@@ -117,6 +118,12 @@ STATUS signAwsRequestInfo(PRequestInfo pRequestInfo)
 
     CHK(pRequestInfo != NULL && pRequestInfo->pAwsCredentials != NULL, STATUS_NULL_ARG);
 
+    // signAwsRequestInfo is a public api function, if service is not specified default to "kinesisvideo" so no breaking changes are introduced to the
+    // api
+    if (pRequestInfo->service[0] == L'\0') {

```

Comment:
Minor but we can use [IS_EMPTY_STRING](https://github.com/awslabs/amazon-kinesis-video-streams-pic/blob/57637ea593f4b43c509413a44d993ed08d7f2616/src/common/include/com/amazonaws/kinesis/video/common/CommonDefs.h#L862) instead.

Reply:
Same below on L179